panther_manager/src/plugins/action/joy_subscription_node.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "panther_manager/plugins/action/joy_subscription_node.hpp"
+
+#include "panther_manager/behavior_tree_utils.hpp"
+
+namespace panther_manager
+{
+
+BT::NodeStatus JoySubscription::onTick(const std::shared_ptr<sensor_msgs::msg::Joy> & last_msg)
+{
+  getInput<std::vector<int>>("buttons", buttons_);
+
+  if (!last_msg) {
+    RCLCPP_ERROR_STREAM(this->logger(), GetLoggerPrefix(name()) << "There is no joy messages!");
+    return BT::NodeStatus::FAILURE;
+  }
```

Comment:
Joy publishers are usually async so it will just fail but for this tree it will finish the Sequence and skip DockRobotAction.
```
      <ForceSuccess>
        <Sequence>
          <JoySubscription topic_name="/panther/joy" buttons="0;0;1;0;0;0;1;1" />
          <DockRobotAction action_name="/panther/dock_robot" use_dock_id="True" dock_id="main_dock"
            dock_type="panther_docking_panther_charging_dock"
            navigate_to_staging_pose="False" />
        </Sequence>
      </ForceSuccess>
```